commands/FBDisplayCommands.py

```diff
@@ -40,21 +54,43 @@ def args(self):
   def options(self):
     return [
       fb.FBCommandArgument(short='-c', long='--color', arg='color', type='string', default='red', help='A color name such as \'red\', \'green\', \'magenta\', etc.'),
-      fb.FBCommandArgument(short='-w', long='--width', arg='width', type='CGFloat', default=2.0, help='Desired width of border.')
+      fb.FBCommandArgument(short='-w', long='--width', arg='width', type='CGFloat', default=2.0, help='Desired width of border.'),
+      fb.FBCommandArgument(short='-d', long='--depth', arg='depth', type='int', default=0, help='Number of levels of subviews to border. Each level gets a different color beginning with the provided or default color'),
     ]
 
   def run(self, args, options):
-    colorClassName = 'UIColor'
-    isMac = runtimeHelpers.isMacintoshArch()
+    def setBorder(layer, width, color, colorClass):
+      lldb.debugger.HandleCommand('expr (void)[%s setBorderWidth:(CGFloat)%s]' % (layer, width))
+      lldb.debugger.HandleCommand('expr (void)[%s setBorderColor:(CGColorRef)[(id)[%s %sColor] CGColor]]' % (layer, colorClass, color))
 
+    depth = int(options.depth)
+    isMac = runtimeHelpers.isMacintoshArch()
+    color = options.color
+    assert color in self.colors, "Color must be one of the following: {}".format(" ".join(self.colors))
+    colorClassName = 'UIColor'
     if isMac:
       colorClassName = 'NSColor'
 
-    layer = viewHelpers.convertToLayer(args[0])
-    lldb.debugger.HandleCommand('expr (void)[%s setBorderWidth:(CGFloat)%s]' % (layer, options.width))
-    lldb.debugger.HandleCommand('expr (void)[%s setBorderColor:(CGColorRef)[(id)[%s %sColor] CGColor]]' % (layer, colorClassName, options.color))
+    if viewHelpers.isUIView(args[0]):
+      prevLevel = 0
+      for view, level in viewHelpers.subviewsOfView(args[0]):
+        if level > depth:
+           break
+        if prevLevel != level:
+          color = self.nextColorAfterColor(color)
+          prevLevel = level
+        layer = viewHelpers.convertToLayer(view)
+        setBorder(layer, options.width, color, colorClassName)
+    else:
+      assert depth <= 0, "Recursive bordering is only supported for UIViews"
```

Comment:
There might be a misunderstanding here. The purpose of this assert is to make sure recursive bordering is not used for layers (or anything that's not a UIView or NSView). However regular bordering is still permitted for layers. So if you're in the code path that takes you to this else block, we should validate that `depth` is not provided to make sure recursive bordering won't happen.